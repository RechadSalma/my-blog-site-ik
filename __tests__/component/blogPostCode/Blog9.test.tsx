import { render } from "@testing-library/react";
import {
  ReactHOC,
  ReactComponent,
  UseHOC,
} from "../../../components/blogPostCodes/Blog9";

describe("<Blog9/> code for blog9 component", () => {
  // const blogNineComponent = () => render(<Blog9Codes />);

  test("ReactHOC renders the <code> tag", () => {
    const { getByRole } = render(<ReactHOC />);

    const codeTag = getByRole("code");

    expect(codeTag).toBeInTheDocument();
  });

  // snapshot component
  test("snapshot ReactHOC code tag", () => {
    const { container } = render(<ReactHOC />);

    expect(container).toMatchSnapshot();
  });

  test("ReactComponent renders the <code> tag", () => {
    const { getByRole } = render(<ReactComponent />);

    const codeTag = getByRole("code");

    expect(codeTag).toBeInTheDocument();
  });

  // snapshot component
  test("snapshot ReactComponent code tag", () => {
    const { container } = render(<ReactComponent />);

    expect(container).toMatchSnapshot();
  });

  test("UseHOC renders the <code> tag", () => {
    const { getByRole } = render(<UseHOC />);

    const codeTag = getByRole("code");

    expect(codeTag).toBeInTheDocument();
  });

  // snapshot component
  test("snapshot UseHOC code tag", () => {
    const { container } = render(<UseHOC />);

    expect(container).toMatchSnapshot();
  });
});
