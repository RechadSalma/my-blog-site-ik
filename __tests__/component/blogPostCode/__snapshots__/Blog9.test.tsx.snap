// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Blog9/> code for blog9 component snapshot ReactComponent code tag 1`] = `
<div>
  <pre
    class="preCode mx-auto"
  >
    <code
      class="codeTag mx-auto"
      role="code"
    >
      <div
        class="comment"
      >
        // Normal react component
      </div>
      <div>
        <span
          class="keyword"
        >
          import
        </span>
         React 
        <span
          class="keyword"
        >
          from
        </span>
         "react";
      </div>
      <br />
      <div>
        <span
          class="keyword"
        >
          const 
        </span>
        Component = (arg) =&gt; {
      </div>
      <div
        class="tab1"
      >
        <span
          class="keyword"
        >
          return
        </span>
         (
      </div>
      <div
        class="tab2"
      >
        &lt;div&gt;{arg}&lt;/div&gt;
      </div>
      <div
        class="tab1"
      >
        );
      </div>
      <div>
        };
      </div>
      <br />
      <div>
        <span
          class="keyword"
        >
          export default 
        </span>
        Component;
      </div>
    </code>
  </pre>
</div>
`;

exports[`<Blog9/> code for blog9 component snapshot ReactHOC code tag 1`] = `
<div>
  <pre
    class="preCode mx-auto"
  >
    <code
      class="codeTag mx-auto"
      role="code"
    >
      <div
        class="comment"
      >
        // HOC function
      </div>
      <div>
        <span
          class="keyword"
        >
          export default function
        </span>
         Hoc(Component) {
      </div>
      <div
        class="tab1"
      >
        <span
          class="keyword"
        >
          function
        </span>
         ReactComponent() {
      </div>
      <div
        class="tab2"
      >
        <span
          class="keyword"
        >
          return
        </span>
         &lt;Component /&gt;
      </div>
      <div
        class="tab1"
      >
        <div>
          };
        </div>
        <br />
        <div>
          <span
            class="keyword"
          >
            return
          </span>
           ReactComponent;
        </div>
      </div>
      <div>
        };
      </div>
    </code>
  </pre>
</div>
`;

exports[`<Blog9/> code for blog9 component snapshot UseHOC code tag 1`] = `
<div>
  <pre
    class="preCode mx-auto"
  >
    <code
      class="codeTag mx-auto"
      role="code"
    >
      <div
        class="comment"
      >
        // How to use the HOC function we created
      </div>
      <div>
        <span
          class="keyword"
        >
          const 
        </span>
        HOC = Hoc(() =&gt; &lt;Component arg={hello world} /&gt;);
      </div>
      <br />
      <div
        class="comment"
      >
        // You can now use HOC as JSX component
      </div>
      <div>
        &lt;HOC /&gt;
      </div>
    </code>
  </pre>
</div>
`;
